### ISSUE TITLE
Zookeeper - 7.5.4 - fresh install - ERROR LearnerHandler Unexpected exception

### ISSUE THREAD

**Issue by AGirin** [bitnami/charts#8538]
**Which chart**:
zookeeper:
App Version: 3.7.0
Package Version: 7.5.4

**Describe the bug**
Fresh install on AWS EKS. Zookeeper constantly logs error:
```
2021-12-30 15:42:14,439 [myid:2] - ERROR [LearnerHandler-/xxx.xxx.xxx.xxx:55509:LearnerHandler@715] - Unexpected exception in LearnerHandler: 

java.io.EOFException
	at java.base/java.io.DataInputStream.readInt(DataInputStream.java:397)
	at org.apache.jute.BinaryInputArchive.readInt(BinaryInputArchive.java:96)
	at org.apache.zookeeper.server.quorum.QuorumPacket.deserialize(QuorumPacket.java:86)
	at org.apache.jute.BinaryInputArchive.readRecord(BinaryInputArchive.java:134)
	at org.apache.zookeeper.server.quorum.LearnerHandler.run(LearnerHandler.java:468)
```
**To Reproduce**

Used Kubeapps to deploy to AWS EKS (version 1.21). 3 replicas... LoadBalancer... Zookeeper works fine as far as I can tell but we constantly getting error above... And this is probably not good. 

**Expected behavior**
No errors on fresh install. 

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:16:20Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```

**Comment by miguelaeh** [comment]
Hi @AGirin ,
It seems to be an end-of-file exception, is it possible that Zookeeper is maintaining a corrupted file?

There is a known error in zookeeper that produces a similar trace to this one, but it is not exactly the same, the error is also a `java.io.EOFException` but it is produced by an empty log file, so maybe the same is happening to you with some other file.

**Comment by AGirin** [comment]
@miguelaeh This is fresh install - out of the box so to speak. Nothing was sent to Zookeeper. This error comes up immediately after cluster becomes available...

**Comment by miguelaeh** [comment]
Hi @AGirin ,
I was able to deploy the version you are using without issues, also checked in our internal pipeline that all the tests are passing properly.
Is there any other information about the zookeeper load that you can share?

**Comment by AGirin** [comment]
Not really :). I just pshed Helm chart to our EKS clsuter... It deploys fine... And it works as well... But we are getting those error messages constantly... Not sure if they affect anything... 

When I say it works fine - I mean we can creats nodes etc... Just for test... We hvae no load on zookeeper at all... As we are just researching it...

**Comment by miguelaeh** [comment]
it is weird, does it happen every time you deploy a fresh release? If you delete everything, including PVC, and deploy it again does the error persist?

**Comment by dangets** [comment]
For future finders - I was getting this same error.  I tracked it down to configuring the service as a NLB load balancer, and the load balancer default auto configured some healthcheck listeners for all ports (2181, 2888, & 3888).  It was the health checks on 2888 & 3888 that were causing all of the errors and noise.

2888 is only listened to by the leader (lined up with only 1 node being healthy)
3888 is used during leader election cycles

I fixed this by explicitly setting loadbalancer healthcheck annotation to the below:
```
      service.beta.kubernetes.io/aws-load-balancer-healthcheck-protocol: TCP  # explicitly set healthcheck to only tcp:2181 to prevent auto-creation of other ports 2888 & 3888
      service.beta.kubernetes.io/aws-load-balancer-healthcheck-port: 2181 
```
